src/Http/Response.php

```diff
@@ -1197,6 +1197,62 @@ public function withAddedLink(string $url, array $options = [])
         return $this->withAddedHeader('Link', '<' . $url . '>' . $param);
     }
 
+    /**
+     * Get an array representation of the HTTP Link header values
+     *
+     * @return array
+     */
+    public function getParsedLink(): array
+    {
+        $linkHeader = $this->getHeader('Link');
+        $result = [];
+        foreach ($linkHeader as $link) {
+            $result[] = $this->extractLinkHeaderData($link);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Parses one item of the HTTP link header into an array
+     *
+     * @param string $linkHeader The HTTP Link header part
+     * @return array[]
+     */
+    protected function extractLinkHeaderData(string $linkHeader)
+    {
+        preg_match('/<(.*)>[; ]?[; ]?(.*)?/', $linkHeader, $matches);
+
+        $url = $matches[1];
+        $parsedParams = [];
+
+        $params = $matches[2];
+        if ($params) {
+            $explodedParams = explode(';', $params);
+            foreach ($explodedParams as $param) {
+                $explodedParam = explode('=', $param);
+                $parsedParams[trim($explodedParam[0])] = trim($explodedParam[1], '"');
+            }
+        }
+
+        if (isset($parsedParams['rel'])) {
+            $rel = $parsedParams['rel'];
+            unset($parsedParams['rel']);
+
+            return [
+                $rel => [
```

Comment:
Not sure about giving special treatment to the `rel` attribute, especially when it's optional.

Reply:
My reasoning was the fact, that most of the API's I know use exactly that `rel` attribute to identify what that specific link is for. So users would have an easier time to get exactly the link they want instead of having to iterate over the simple array again and manually checking the rel attribute.

But if this is not the way we want to go then a simple array structure is also fine by me.

Reply:
> So users would have an easier time to get exactly the link they want instead of having to iterate over the simple array again and manually checking the rel attribute.

Currently you still need to iterate over the array returned by `getParsedLink()` method.

Reply:
Right... My internal implementation made it possible to do something like
```
$new->getParsedLink()['next']['url'];
```
But I somehow mangled that up here when I transfered that to Cake. 
Anyways - if we don't want to treat the `rel` attribute as a sort of key for that specific Link entry then this is a bad idea anyways.

Reply:
> $new->getParsedLink()['next']['url'];

You can't have an array structure which makes above usage possible since `rel` isn't always unique.

Reply:
The rel "key behavior" has now been removed.